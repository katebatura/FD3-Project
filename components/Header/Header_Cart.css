
.header__cart .fa-shopping-cart {
    color: black;
    font-size: 36px;
    position: relative;
    padding: 10px;
}

.cart-qty {
    position: absolute;
    top: 5px;
    right: 0;
    width: 25px;
    height: 25px;
    border-radius: 50%;
    background-color:  #19bde8;
    color: white;
    text-align: center;
    line-height: 25px;
    font-size: 14px;
    font-weight: bold;
}