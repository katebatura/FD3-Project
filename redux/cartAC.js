const QTY_ADD = 'QTY_ADD';
const PROD_ADD = 'PROD_ADD';
const PROD_DEL = 'PROD_DEL';
const SUM_QTY = 'SUM_QTY';
const DELIVERY_CHNG = 'DELIVERY_CHNG';

const prod_add = function(prodid, info) {
  return {
    type: PROD_ADD,
    prodid: prodid,
    info: info,
  };
}

const prod_del = function(prodid) {
  return {
    type: PROD_DEL,
    prodid: prodid,
  };
}

const qty_add = function(prodid, newQTY, newSum) {
  return {
    type: QTY_ADD,
    prodid: prodid,
    newQTY: newQTY,
    newSum: newSum
  };
}

const sum_qty = function(qty) {
  return {
    type: SUM_QTY,
    qty: qty,
  }
}

const delivery_chng = function(num) {
  return {
    type: DELIVERY_CHNG,
    num: num,
  }
}

export {
  prod_add, PROD_ADD,
  prod_del, PROD_DEL,
  qty_add, QTY_ADD,
  sum_qty, SUM_QTY,
  delivery_chng, DELIVERY_CHNG,
}
