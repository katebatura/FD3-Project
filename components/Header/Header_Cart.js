"use strict";

import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import {connect} from 'react-redux';

import { clearLocalStorage } from '../../LocalStorage/LocalStorage'

import './Header_Cart.css';

class Header_Cart extends React.PureComponent {
 
  static propTypes = {
    selectedProducts: PropTypes.object.isRequired, // передано из Redux
  };

  componentWillReceiveProps = (newProps) => {
    console.log(newProps)
    if(newProps.selectedProducts.productsQTY == 0) {
      clearLocalStorage();}
  };

  render() {

    return (
    <div className="header__cart">
      <NavLink to="/cart" className="breadcrumbs"> 
        <i className="fa fa-shopping-cart">
        {
          this.props.selectedProducts.productsQTY != 0 ?
          <div className = "cart-qty">{this.props.selectedProducts.productsQTY}</div> : null
        }
      </i>
      </NavLink>
    </div>
    );

  }

}

const mapStateToProps = function (state) {
  return {
    // весь раздел Redux state под именем counters будет доступен
    // данному компоненту как this.props.counters
    selectedProducts: state.selectedProducts,
  };
};

export default connect(mapStateToProps)(Header_Cart);