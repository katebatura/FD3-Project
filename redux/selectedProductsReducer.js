import { SUM_QTY } from './cartAC';

const initState = {
    
    productsQTY: 0,
  
  }
  
  
  function selectedProductsReducer(state = initState, action) {
    switch (action.type) {

      case SUM_QTY: { 
        let newQTY = state.productsQTY + action.qty;
        let newState = {...state,
          productsQTY:  newQTY};
        return newState;
      }

      default:
        return state;
    }
  }
  
  export default selectedProductsReducer;