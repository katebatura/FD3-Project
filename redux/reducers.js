import { combineReducers } from 'redux';

import productsListReducer from "./productsListReducer";
import cartReducer from "./cartReducer";
import selectedProductsReducer from "./selectedProductsReducer";

let combinedReducer = combineReducers({
    // редьюсер countersReducer отвечает за раздел state под именем counters
    products: productsListReducer,
    cart: cartReducer, 
    selectedProducts: selectedProductsReducer,
});

export default combinedReducer;
