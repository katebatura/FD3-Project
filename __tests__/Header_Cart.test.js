"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import { createStore } from 'redux';
import combinedReducer from '../redux/reducers.js';
import { Provider } from 'react-redux';

import Header_Cart from '../components/Header/Header_Cart';


let store=createStore(combinedReducer);

test('внешний вид Header_Cart', () => {

  // создаём тестовую версию компонента
  const component = renderer.create(
    <Router>
      <Provider store={store}>
        <Header_Cart />
      </Provider>
    </Router>
  );

  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

      
});
