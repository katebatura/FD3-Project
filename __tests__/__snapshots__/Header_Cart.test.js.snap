// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`внешний вид Header_Cart 1`] = `
<div
  className="header__cart"
>
  <a
    aria-current={null}
    className="header__cart-link"
    href="/cart"
    onClick={[Function]}
    style={undefined}
  >
    <i
      className="fas fa-cart-arrow-down"
    />
  </a>
</div>
`;
